dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/FileExt.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+@file:Suppress("TooManyFunctions")
+package com.datadog.android.core.internal.persistence.file
+
+import com.datadog.android.core.internal.utils.sdkLogger
+import java.io.File
+import java.io.FileFilter
+import java.io.FilenameFilter
+
+/*
+ * The java.lang.File class throws a SecurityException for the following calls:
+ * - canRead()
+ * - canWrite()
+ * - delete()
+ * - exists()
+ * - isFile()
+ * - isDir()
+ * - listFiles(…)
+ * - length()
+ * The following set of extension make sure that every call to those methods
+ * is safeguarded to avoid crashing the customer's app.
+ */
+
+private fun <T> File.safeCall(
```

Comment:
shouldn't we use these methods also in the new `FileHandler` implementation ?

Reply:
The new `FileHandler` class already uses `try`/`catch` blocks because it also uses Input/Output streams. So there's no need to nest try catch blocks